### ISSUE TITLE
Promote kueue apis to V1

### ISSUE THREAD

**Issue by kannon92** [kubernetes-sigs/kueue#3476]
**What would you like to be added**:
Promote the core Kueue APIs to V1.
**Why is this needed**:
We would need to use Kueue in more of our products. Many customers would like a stable API to build on top of.

We would like to start the discussion for what it would take to promote the v1beta1(2) APIs to v1 and stablize the API.
**Completion requirements**:

This enhancement requires the following artifacts:

- [ ] Design doc
- [ ] API change
- [ ] Docs update

The artifacts should be linked in subsequent comments.

**Comment by kannon92** [comment]
It is not clear to me if https://github.com/kubernetes-sigs/kueue/issues/768 must be satisfied before we promote to V1 or not.

**Comment by mimowo** [comment]
Actually, we use in Kueue both alpha and beta.

I'm wondering if we could migrate to v1 only a subset of API which we consider already stable (like Workload, LQ, CQ and so on), while keeping the other still in alpha or beta (like Topology or MultiKueue, or other new features).

This would give us more flexibility and serial that we promote APIs which aren't battle tested yet. In any case new features to Kueue will start in Alpha.
 
WDYT @kannon92 @tenzen-y ?

**Comment by mimowo** [comment]
> It is not clear to me if https://github.com/kubernetes-sigs/kueue/issues/768 must be satisfied before we promote to V1 or not.

This is whitelist for v1beta2, so yes, this should happen before  v1

**Comment by kannon92** [comment]
> This would give us more flexibility and serial that we promote APIs which aren't battle tested yet. In any case new features to Kueue will start in Alpha.

I don't think we need features to start in alpha API fields. Unless we are introducing a new API field entirely. One can always add a new field to v1 or v1beta API.

**Comment by kannon92** [comment]
> This is whitelist for v1beta2, so yes, this should happen before v1

My concern is not there isn't really a hard and fast rule on this one so I worry that it will stay as a wishlist with this API in beta while we add more items to it.

**Comment by mimowo** [comment]
When I was talking about new features I was thinking that the API objects, like recently introduced Topology or Cohort, should go via API. For extending existing objects, sure we have no choice than extend them.

As for the whitelist, I'm with you, I would like to promote the APIs, but the list is long and so think it will be safer and with less friction if we move via interim step of v1beta2 than directly to v1.

**Comment by tenzen-y** [comment]
As I commented in https://github.com/kubernetes-sigs/kueue/issues/3192#issuecomment-2396510965, I would like to wait any API graduation (even v1beta2) for the graduation of the TAS and HirarchyCohort since both are still alpha stage and potentially plan API change.

**Comment by kannon92** [comment]
> I'm wondering if we could migrate to v1 only a subset of API which we consider already stable (like Workload, LQ, CQ and so on), while keeping the other still in alpha or beta (like Topology or MultiKueue, or other new features).

@tenzen-y is there room to do something like this?

I would really like to see some kind of stabilization of the API for at least the core resources.

**Comment by mimowo** [comment]
I would like to consider that before v1: https://github.com/kubernetes-sigs/kueue/issues/8607 which relates to https://github.com/kubernetes-sigs/kueue/issues/7035
